Ruby 'leaked symbols' checker does not expect dtrace related symbols to exist.

This was offered and  merged upstream:
https://bugs.ruby-lang.org/issues/20910
https://github.com/ruby/ruby/commit/8149f4d6abd4df27d07c047b133451a1364f8d9a

From 8149f4d6abd4df27d07c047b133451a1364f8d9a Mon Sep 17 00:00:00 2001
From: Nobuyoshi Nakada <nobu@ruby-lang.org>
Date: Sat, 30 Nov 2024 09:10:29 +0900
Subject: [PATCH] [Bug #20910] dtrace related symbols are not considered leaked

--- a/tool/leaked-globals
+++ b/tool/leaked-globals
@@ -70,6 +70,7 @@ Pipe.new(NM + ARGV).each do |line|
   next unless n.sub!(/^#{SYMBOL_PREFIX}/o, "")
   next if n.include?(".")
   next if !so and n.start_with?("___asan_")
+  next if !so and n.start_with?("__dtrace")
   case n
   when /\A(?:Init_|InitVM_|pm_|[Oo]nig|dln_|coroutine_)/
     next
